### DISCUSSION TITLE
Continue statement from within a plugin

### DISCUSSION THREAD

**Discussion by FabianRe1** [jspsych/jsPsych#2602] [Q&A] [answered]
Hello,

I am currently struggling to cope with missing responses in my task. I  create a long timeline in a for loop that I afterwards run. However, in case of missing responses, I would like to jump to the next trial. The rough set up looks like the following:
```
for i in trials:
  var stim1 = {}
  timeline.push(stim1)

  var stim2 = {}
  timeline.push(stim2)

  // if no response although one is expected, I would like to continue with the next iteration of the for loop

  var stim3 = {}
  timeline.push(stim3)

jsPsych.run(timeline)
```
_____

Now, it could happen that I offer a choice in stim 2. If I receive a response, I want to continue as usual. If the participant does not respond, I would like to execute the continue statement and, as a result, not push stim 3. 
Is there any way to do this? Or to have a statement that if I receive a "null" response when I expect a buttonpress to ignore X elements in my timeline?
Any hints on how to handle this are very much appreciated. 

Best wishes,
Fabian

**Comment by jodeleeuw** [comment, marked as answer]
Hi @FabianRe1,

One way to do this kind of thing is to use a timeline's [`conditional_function`](https://www.jspsych.org/7.2/overview/timeline/#conditional-timelines) to evaluate whether the trial should run or not.

In this case you'll still add stim 3 to the timeline, but you'll just only have that part of the timeline run if the right conditions are met. 

In most circumstances you can't conditionally add things to the timeline based on user input because the timeline is generated before `jsPsych.run()` is called and so there is no data available to evaluate the conditional statement. This is why the `conditional_function` approach is normally needed. There is one method [`.addNodeToEndOfTimeline()`](https://www.jspsych.org/7.0/reference/jspsych/#jspsychaddnodetoendoftimeline) that allows for adding to a timeline during the experiment, but it is not especially developer-friendly to use in most circumstances.

**Comment by FabianRe1** [comment]
Hi jodeleeuw,

thanks a lot for your help! I managed to get it to work. 
My actual code is slightly more complicated with several different decisions at different stages per trial.
This worked for me in the end (perhaps useful for others):

1. Push everything up to the first decision point to the timeline.
2. At every potential decision point, check if a decision was presented and whether the response is "null". 
3. If there was a response, put everything in the timeline until the next decision point (including the next condition node).
So basically it is nested until the end of the trial. I will attach the code below, although out of context it might not be as helpful. 
4. Afterwards there is a conditional text that is presented if a response was missed, from there I move on to the next trial. 

Thanks a lot for your help. Should be good to go in the upcoming days! :)

Best wishes,
Fabian 

______

```
      var if_node_2 = {
      timeline: [conditional_stimulus3_padded, fixation_enc, conditional_stimulus3, conditional_fixation, conditional_stimulus4, conditional_fixation_2, reward],
      conditional_function: function(){
          // get the data from the previous trial,
          // and check which key was pressed
          if (jsPsych.data.get().last().values()[0].response== null && switch_tmp){
              return false;
          } else {
              return true;
            }
        }
    }

      var if_node = {
      timeline: [conditional_stimulus2_padded, fixation_enc, stimulus3, if_node_2],
      conditional_function: function(){
          // get the data from the previous trial,
          // and check which key was pressed
          if (jsPsych.data.get().last().values()[0].response== null && switch_tmp1){
              return false;
          } else {
              return true;
            }
        }
    }

    var if_node_no_resp = {
    timeline: [if_node_no_resp_presentation],
    conditional_function: function(){
        // get the data from the previous trial,
        // and check which key was pressed
        if (no_response==1){
            return true;
        } else {
            return false;
          }
      }
  }

    // Push block 1
    timeline.push(fixation_enc_start, stimulus1)
    timeline.push(fixation_enc, stimulus2)
    timeline.push(if_node)
    timeline.push(if_node_no_resp)
```
